pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -1200,17 +1199,23 @@ func (m *kubeGenericRuntimeManager) computeInitContainerActions(pod *v1.Pod, pod
 				}
 
 				// If the init container is in unknown state, restart it.
-				changes.ContainersToKill[status.ID] = containerToKillInfo{
-					name:      container.Name,
-					container: container,
-					message: fmt.Sprintf("Init container is in %q state, try killing it before restart",
-						status.State),
-					reason: reasonUnknown,
-				}
-				changes.InitContainersToStart = append(changes.InitContainersToStart, i)
+				message = fmt.Sprintf("%s is in %q state, try killing it before restart", message, status.State)
+				reason = reasonUnknown
+				restartContainer = true
 			}
 		}
 
+		if restartContainer {
```

Comment:
I really don't like this generalization. The switch above is so huge that it's hard for me to confirm for sure all branches will get here with all the needed arguments without any of them potentially conflicted with each other. Can you please move keep the logic as is, just add a new condition for restart

Reply:
Will keep it as is. I also don't like the switch, it's very hard to follow